### ISSUE TITLE
Handle disk write errors

### ISSUE THREAD

**Issue by kumavis** [MetaMask/metamask-extension#4145]
https://sentry.io/metamask/metamask/issues/505549706/events/19184253266/

> IO error: .../000003.log: FILE_ERROR_FAI…

this likely means the user's disk is full. not much we can do about that, but we should handle the error and show a warning to the user

**Comment by cjeria** [comment]
@kumavis How many users are having this problem? This'll help me organize high priority tasks with low priority ones.

**Questions to help design a solution:**
Can the user still perform actions with MetaMask? 
Can the user do anything to solve the problem (ie, clear cache, delete old transactions)?

**Possible solutions**
Universal error message component to display in the UI.

**Comment by kumavis** [comment]
@cjeria pretty low, but falls into a still-low-occurrence-but-generally-useful category of show-stopping alerts for the user

**Comment by danfinlay** [comment]
This error is for incorrect password, so it isn't a true error.

_However_, if this error is breaking our storage strategy, then _that_ is a serious issue.

We recently encountered a bug where an error thrown within an event-emitter handler would stop the emitter from emitting.

Is it possible that an error as simple as "incorrect password" could prevent MetaMask from writing to disk, breaking its storage stream?

**Comment by bdresser** [comment]
let's file a separate issue for general error display - this doesn't need to be blocked on design

**Comment by danfinlay** [comment]
Possibly related to #4287

**Comment by bdresser** [comment]
possibly fixed by #4570

**Comment by danfinlay** [comment]
I don't think this is fixed by #4570. While that PR fixes one type of storage bug, this issue here is about showing when other errors happen.

**Comment by bdresser** [comment]
probably best thing to do is tell the user to back up their seed phrase. might happen in background, so maybe a browser notif

**Comment by Gudahtt** [comment]
I'm re-adding the "needsDesign" label because this still needs a design, but I agree that the design shouldn't block fixing this. We can create something "good enough" and improve the design later if need be.

**Comment by Gudahtt** [comment]
Note that there is a second place where we persist state during initialization after running the state through the migrator where we also need to catch this error.
